code/modules/xenoarcheaology/effect.dm

```diff
@@ -3,19 +3,45 @@
 	var/effect = EFFECT_TOUCH
 	var/effectrange = 4
 	var/trigger = TRIGGER_TOUCH
-	var/atom/holder
+	var/datum/component/artifact_master/master
 	var/activated = 0
-	var/chargelevel = 0
+	var/chargelevel = 1
 	var/chargelevelmax = 10
 	var/artifact_id = ""
 	var/effect_type = 0
 
-/datum/artifact_effect/New(var/atom/location)
+	var/req_type = /atom/movable
+
+	var/image/active_effect
+	var/effect_icon = 'icons/effects/effects.dmi'
+	var/effect_state = "sparkles"
+	var/effect_color = "#ffffff"
+
+// The last time the effect was toggled.
+	var/last_activation = 0
+
+/datum/artifact_effect/Destroy()
+	if(master)
+		master = null
+
 	..()
-	holder = location
+
+/datum/artifact_effect/New(var/datum/component/artifact_master/newmaster)
+	..()
+	if(!newmaster)
+		qdel(src)
```

Comment:
Hmmmm not sure about this. Letting it runtime would probably be more informative than (for example) something trying to create an instance, getting a qdeleted instance, and keeping a reference to it anyway.